addons/html_editor/static/src/editor/link/link_plugin.js

```diff
@@ -91,12 +97,23 @@ export class LinkPlugin extends Plugin {
     toggleLinkTools(options = {}) {
         // ...
         // const shouldFocusUrl = options.shouldFocusUrl === undefined ? true : options.shouldFocusUrl;
-        const linkEl = findInSelection(this.shared.getEditableSelection(), "a");
-        if (
-            linkEl &&
-            (!linkEl.matches(this.customizableLinksSelector) || !linkEl.isContentEditable)
-        ) {
-            return;
+        // const linkEl = findInSelection(this.shared.getEditableSelection(), "a");
+        // if (
+        //     linkEl &&
+        //     (!linkEl.matches(this.customizableLinksSelector) || !linkEl.isContentEditable)
+        // ) {
+        //     return;
+        // }
+
+        const selection = this.shared.getEditableSelection();
+        const linkElement = closestElement(selection.anchorNode, "A");
+        if (!this.overlay && linkElement) {
+            this.overlay = this.shared.createOverlay(LinkPopover, {
```

Comment:
the overlay should be created once, and then open/close depending on the situation